ctru-rs/src/services/cam.rs

```diff
@@ -0,0 +1,1050 @@
+        camera: CamSelect,
+        x: i16,
+        y: i16,
+        width: i16,
+        height: i16,
+    ) -> crate::Result<()> {
+        unsafe {
+            let r = ctru_sys::CAMU_SetAutoWhiteBalanceWindow(camera.bits(), x, y, width, height);
+            if r < 0 {
+                Err(r.into())
+            } else {
+                Ok(())
+            }
+        }
+    }
+
+    pub fn set_noise_filter(&self, camera: CamSelect, enabled: bool) -> crate::Result<()> {
+        unsafe {
+            let r = ctru_sys::CAMU_SetNoiseFilter(camera.bits(), enabled);
+            if r < 0 {
+                Err(r.into())
+            } else {
+                Ok(())
+            }
+        }
+    }
+
+    pub fn synchronize_vsync_timing(
+        &self,
+        camera1: CamSelect,
+        camera2: CamSelect,
+    ) -> crate::Result<()> {
+        unsafe {
+            let r = ctru_sys::CAMU_SynchronizeVsyncTiming(camera1.bits(), camera2.bits());
+            if r < 0 {
+                Err(r.into())
+            } else {
+                Ok(())
+            }
+        }
+    }
+
+    pub fn get_latest_vsync_timing(&self, port: CamPort, past: u32) -> crate::Result<i64> {
+        let mut timing = 0;
+        unsafe {
+            let r = ctru_sys::CAMU_GetLatestVsyncTiming(&mut timing, port.bits(), past);
+            if r < 0 {
+                Err(r.into())
+            } else {
+                Ok(timing)
+            }
+        }
+    }
+
+    pub fn write_register_i2c(&self, camera: CamSelect, addr: u16, data: u16) -> crate::Result<()> {
+        unsafe {
+            let r = ctru_sys::CAMU_WriteRegisterI2c(camera.bits(), addr, data);
+            if r < 0 {
+                Err(r.into())
+            } else {
+                Ok(())
+            }
+        }
+    }
+
+    pub fn write_mcu_variable_i2c(
+        &self,
+        camera: CamSelect,
+        addr: u16,
+        data: u16,
+    ) -> crate::Result<()> {
+        unsafe {
+            let r = ctru_sys::CAMU_WriteMcuVariableI2c(camera.bits(), addr, data);
+            if r < 0 {
+                Err(r.into())
+            } else {
+                Ok(())
+            }
+        }
+    }
+
+    pub fn read_register_i2c_exclusive(&self, camera: CamSelect, addr: u16) -> crate::Result<u16> {
+        unsafe {
+            let mut data = 0;
+            let r = ctru_sys::CAMU_ReadRegisterI2cExclusive(&mut data, camera.bits(), addr);
+            if r < 0 {
+                Err(r.into())
+            } else {
+                Ok(data)
+            }
+        }
+    }
+
+    pub fn read_mcu_variable_i2c_exclusive(
+        &self,
+        camera: CamSelect,
+        addr: u16,
+    ) -> crate::Result<u16> {
+        unsafe {
+            let mut data = 0;
+            let r = ctru_sys::CAMU_ReadMcuVariableI2cExclusive(&mut data, camera.bits(), addr);
+            if r < 0 {
+                Err(r.into())
+            } else {
+                Ok(data)
+            }
+        }
+    }
```

Comment:
Do you think these functions might need to be marked unsafe?

Reply:
In favor of removing these possibly because they may not have much use and the unsafe factor. Thoughts?

Reply:
Yeah, it might be better to remove them since they seem unsafe and probably aren't used very often. The wrapper is simple so it shouldn't be hard to call the ctru_sys functions directly.